vignettes/epidist.Rmd

```diff
@@ -1,201 +1,211 @@
 
-```{r observe-data}
-truncated_obs <- obs |>
-  filter_obs_by_obs_time(obs_time = 25)
-truncated_obs <- truncated_obs[sample(1:.N, 200, replace = FALSE)]
+```{r delay, fig.cap="(ref:delay)"}
+obs[case %% 50 == 0, ] |>
+  ggplot(aes(y = case)) +
+  geom_segment(
+    aes(x = ptime, xend = stime, y = case, yend = case),
+    col = "grey"
+  ) +
+  geom_point(aes(x = ptime), col = "#56B4E9") +
+  geom_point(aes(x = stime), col = "#009E73") +
+  labs(x = "Event time (day)", y = "Case number") +
+  theme_minimal()
 ```
 
-Plot primary cases (columns), and secondary cases (dots) by the observation time of their secondary events. This reflects would could be observed in real-time.
-
-```{r observed-cases}
-truncated_cases <- construct_cases_by_obs_window(
-  obs, windows = c(25), obs_type = "stime"
-)
+`obs` is a `data.table` object with additional columns for the delay `delay` and time of secondary event `stime`.
+The secondary event time is simply the primary event time plus the delay:
 
-plot_cases_by_obs_window(truncated_cases)
+```{r}
+all(obs$ptime + obs$delay == obs$stime)
 ```
 
-We can alternatively plot the observed data based on primary events. This corresponds to a retrospective cohort based view of the data.
+If we received the data `obs` as above then estimating the delay distribution would be easy, and this package wouldn't need to exist.
+However, in reality, we will almost never receive the data as above.
 
-```{r cohort-observed-cases}
-truncated_cases <- construct_cases_by_obs_window(
-  obs, windows = c(25), obs_type = "ptime"
-)
+First, the times of primary and secondary events will usually be censored.
+This means that rather than exact event times, we observe event times within an interval.
+Here we suppose that the interval is daily, resulting in an obscuring of the exact delay data (Figure \@ref(fig:cens)):
 
-plot_cases_by_obs_window(truncated_cases)
+```{r}
+# observe_process() should be renamed and include a "daily" i.e. 1 argument
+obs_cens <- obs |> observe_process()
 ```
 
-Plot the true continuous delay distribution and the empirical observed distribution for each observation window.
+(ref:cens) Interval censoring of the primary and secondary event times obscures the delay times.
 
-```{r empirical-dist}
-combined_obs <- combine_obs(truncated_obs, obs)
-
-plot_empirical_delay(
-  combined_obs, meanlog = secondary_dist$meanlog[[1]],
-  sdlog = secondary_dist$sdlog[[1]]
-)
+```{r cens, fig.cap="(ref:cens)"}
+ggplot(obs_cens, aes(x = delay, y = delay_daily)) +
+  geom_point(col = "#E69F00") +
+  geom_abline(slope = 1, intercept = 0, linetype = "dashed", col = "grey") +
+  theme_minimal() +
+  coord_fixed() +
+  labs(x = "Exact delay time (days)", y = "Censored delay time (days)")
 ```
 
-Plot the mean difference between continuous and discrete event time:
+Next, during an outbreak we will usually be estimating delays in real time.
```

Comment:
Epidemiological delays are often ascertained via the secondary event (e.g. when estimating an incubation period the symptom onset event is much more likely to be observed and reported than the infection event).

Reply:
> **What happens if we had the symptom onset date for an individual but not the infection date? Discard their data? Then it's a non-representative sample?

Yes exactly (so no the methods here are not accounting for this) - `epinowcast` has a module for this (the missingness module) but its a tricky problem)

Reply:
Worth mentioning in this vignette that we don't deal with it then? Or leave that to something more advanced? I'd suggest leave to more advanced.

Reply:
I think very briefly pointing it out (and at some later date linking to something more complex) might be a good user journey

Reply:
I've tried something brief about this. Hard to know where exactly to put it.